packages/backend/src/core/discovery/proxies/auto/StarkWareProxy.ts

```diff
@@ -95,31 +73,26 @@ export async function detectStarkWareProxy(
   if (implementation === EthereumAddress.ZERO) {
     return
   }
-  const [
-    callImplementation,
+  const [callImplementation, upgradeDelay, isFinal, proxyGovernance] =
+    await Promise.all([
+      getCallImplementation(provider, address),
+      getUpgradeDelay(provider, address),
+      getFinalizedState(provider, address),
+      getProxyGovernance(provider, address),
+    ])
+
+  const diamond = await getStarkWareDiamond(
```

Comment:
are you sure this method will not throw?

Reply:
That's a good question. It can. Let me fix it.